### PR TITLE
Optimize build times

### PR DESCRIPTION
**Pull request by CraigSiemens**
There are three main changes in this PR.

## Only run SwiftFormat once per file (1st commit)

Previously it was run once per renderer used. If a renderer called another renderer (in the case of nested types) the nested renderer would format the code, then the enclosing renderer would call it again. This would cause waisted work by trying to format code that's already been formatted.

### Results

~17% faster than `main`.

```
Benchmark 1: Branch: main
  Time (mean ± σ):     11.154 s ±  0.566 s    [User: 47.793 s, System: 13.385 s]
  Range (min … max):   10.415 s … 11.865 s    5 runs
  
Benchmark 2: Branch: any-coding-key
  Time (mean ± σ):      9.542 s ±  0.342 s    [User: 49.807 s, System: 16.590 s]
  Range (min … max):    9.326 s … 10.133 s    5 runs
  
Summary
  'Branch: any-coding-key' ran
    1.17 ± 0.07 times faster than 'Branch: main'
```

## Generate the Codable implementation (2nd and 3rd commit) (part of #145)

On its own this change doesn't improve the build time, if anything it makes it worse. It's required for the following change.

## Use one type for all coding keys (4th and 5th commit) (part of #145)

This removes around half of the types being compiled, since every model has keys either generated by the compiler or Bagbutik. This requires the previous change because the swift compiler can only generate the methods if `CodingKeys` is an enum.

### Results

~67% faster than `main`.

```
Benchmark 1: Branch: main
  Time (mean ± σ):     110.734 s ±  0.213 s    [User: 245.244 s, System: 61.398 s]
  Range (min … max):   110.368 s … 111.050 s    10 runs
 
Benchmark 2: Branch: any-coding-key
  Time (mean ± σ):     66.478 s ±  0.168 s    [User: 173.876 s, System: 27.938 s]
  Range (min … max):   66.271 s … 66.743 s    10 runs
 
Summary
  'Branch: any-coding-key' ran
    1.67 ± 0.01 times faster than 'Branch: main'
```

### CODE REVIEW

File: Sources/BagbutikGenerator/Renderers/ObjectSchemaRenderer.swift [hunk 1]
```diff
@@ -62,39 +62,37 @@ public class ObjectSchemaRenderer: Renderer {
                 structContent.append(renderInitializer(parameters: deprecatedPublicInitParameters, deprecated: true, content: { createInitContent(propertiesInfo.deprecatedPublicInitPropertyNames) }))
             }
             structContent.append(renderInitializer(parameters: propertiesInfo.publicInitParameters, content: { createInitContent(propertiesInfo.publicInitPropertyNames) }))
-            if propertiesInfo.needsCustomCoding {
```
Comment (CraigSiemens) [inline Sources/BagbutikGenerator/Renderers/ObjectSchemaRenderer.swift:66]: The following changes look more complicated than they are. This condition was removed causing the indentation of the following lines to change.

File: Sources/BagbutikGenerator/Renderers/ObjectSchemaRenderer.swift [hunk 2]
```diff
@@ -121,7 +119,6 @@ public class ObjectSchemaRenderer: Renderer {
         let encodableProperties: [CodableProperty]
         let decodableProperties: [CodableProperty]
         let hasTypeConstant: Bool
-        let needsCustomCoding: Bool
```
Comment (CraigSiemens) [inline Sources/BagbutikGenerator/Renderers/ObjectSchemaRenderer.swift:125]: This is removed because it's no longer used.

**Comment by MortenGregersen** [comment]
Thank you SO MUCH for this @CraigSiemens! I have been meaning to get back to you on the issue, but kept pushing it... From a quick glance (on my phone) it looks good.

Have you run the tests? I don't see any tests changed.

Also, the spec file should not be included. I will review it more thoroughly one of the coming days (it is 1 AM here).

Thank you again!

**Comment by CraigSiemens** [comment]
😬 I forgot about the tests, I'll look into it.

I'll also add the spec file to the gitignore since I kept accidentally committing it and obviously I missed one.

**Comment by CraigSiemens** [comment]
@MortenGregersen The tests are all updated if you want to take a look at it again.

File: Sources/BagbutikGenerator/Renderers/ObjectSchemaRenderer.swift (hunk 1 above)
Comment (MortenGregersen) [reply Sources/BagbutikGenerator/Renderers/ObjectSchemaRenderer.swift:66]: This is not a problem, when I use GitHub's "Hide whitespace changes" ;)

**Comment by MortenGregersen** [comment]
Closes #145

**Comment by MortenGregersen** [comment]
> @MortenGregersen The tests are all updated if you want to take a look at it again.

Thank you Craig. I can see that the tests are failing on Linux. I think this has something to do with the `XCTestCase` check you did in `Renderer.swift`. I have removed it and explicitly specified that it should format the code in tests. If everything is green now, I will test it in AppDab and merge it in.

**Comment by MortenGregersen** [comment]
> @MortenGregersen The tests are all updated if you want to take a look at it again.

Thank you @CraigSiemens . I can see that the tests are failing on Linux. I think this has something to do with the `XCTestCase` check you did in `Renderer.swift`. I have removed it and explicitly specified that it should format the code in tests. If everything is green now, I will test it in AppDab and merge it in.

**Pull request merged**